### ISSUE TITLE
propagation logic issues

### ISSUE THREAD

**Issue by krchristie** [geneontology/paint#21]
Hi,

I have just annotated PTHR23193 - NUCLEAR PORE COMPLEX PROTEIN NUP 

There seem to be some issues with the propagation logic that are producing results that aren't really what I want.

**Specific experimental annotations blocking propagation of a related but slightly more general term to that sequence.**

For example, MOUSE:NUP214 is already annotated to "mRNA export from nucleus (GO:0006406) by IMP. It is not receiving the more general annotation "RNA export from nucleus (GO:0006405)" by IBA. 

I think it would be appropriate for this more general term to be propagated to these sequences even though they already have more specific experimental annotations.

**NOT annotations that I don't really want to make**

I blocked propagation of two CC terms at the node 9BILA:PTN000573963. I only want to block propagation of the two terms since I think they are too specific for the NUP214 section of the tree. However,  I don't want to be generating NOT annotations to these two terms. I'd just like to not make positive annotations from this point on. In this case, the fact that the tree structure seems odd may be an issue because if I don't annotate to the root, there are sequences that won't get annotations I think are appropriate.

However, more generally, there are many times where the behavior I want is merely to stop propagation, not to generate NOT annotations. There are only a few specific occasions where I would really want to generate NOT annotations. 

-Karen

lines from the GAF showing annotations to the root, and annotations propagated to mouse NUP214

<pre>!gaf-version: 2.0
! Saved by krc using Paint 2.0-beta19 on 08 Jun 2016 using PANTHER 10.0 built on the 2014 GCRP release
PANTHER PTN000573960    PTN000573960            GO:0008139      PAINT_REF:23193 IBD     RGD:3216|RGD:1304977    F                       protein taxon:2759      20160112        GO_Central
PANTHER PTN000573960    PTN000573960            GO:0017056      PAINT_REF:23193 IBD     SGD:S000005624|HGNC:8062|FB:FBgn0061200|RGD:620680      F                       protein taxon:2759      20160112        GO_Central
PANTHER PTN000573960    PTN000573960            GO:0005487      PAINT_REF:23193 IBD     SGD:S000005624|HGNC:8062|FB:FBgn0061200|MGI:MGI:1095411|HGNC:8064       F                       protein taxon:2759      20160112        GO_Central

PANTHER PTN000573960    PTN000573960            GO:0044615      PAINT_REF:23193 IBD     SGD:S000005624|RGD:3216|HGNC:8062       C                       protein taxon:2759      20160607        GO_Central
PANTHER PTN000573960    PTN000573960            GO:0044613      PAINT_REF:23193 IBD     SGD:S000005624|FB:FBgn0061200   C                       protein taxon:2759      20160607        GO_Central

PANTHER PTN000573960    PTN000573960            GO:0006606      PAINT_REF:23193 IBD     PomBase:SPAC30D11.04c|SGD:S000005624|FB:FBgn0061200|MGI:MGI:1095411|FB:FBgn0010660      P                       protein taxon:2759      20160608        GO_Central
PANTHER PTN000573960    PTN000573960            GO:0006405      PAINT_REF:23193 IBD     SGD:S000005624|MGI:MGI:1095411  P                       protein taxon:2759      20160608        GO_Central

MGI     MGI:1095411     Nup214          GO:0008139      PAINT_REF:23193 IBA     PANTHER:PTN000573960|RGD:3216|RGD:1304977       F       nucleoporin 214 UniProtKB:Q80U93|PTN000574025|CAN|D2H9S46E      protein taxon:10090     20160112        GO_Central
MGI     MGI:1095411     Nup214          GO:0017056      PAINT_REF:23193 IBA     PANTHER:PTN000573960|SGD:S000005624|UniProtKB:P49790|FB:FBgn0061200|RGD:620680  F       nucleoporin 214 UniProtKB:Q80U93|PTN000574025|CAN|D2H9S46E      protein taxon:10090     20160112        GO_Central

MGI     MGI:1095411     Nup214  NOT     GO:0044615      PAINT_REF:23193 IBA     PANTHER:PTN000573963    C       nucleoporin 214 UniProtKB:Q80U93|PTN000574025|CAN|D2H9S46E      protein taxon:10090     20160607        GO_Central
MGI     MGI:1095411     Nup214  NOT     GO:0044613      PAINT_REF:23193 IBA     PANTHER:PTN000573963    C       nucleoporin 214 UniProtKB:Q80U93|PTN000574025|CAN|D2H9S46E      protein taxon:10090     20160607        GO_Central
</pre>

**Comment by selewis** [comment]
Yes, "blocking" as opposed to "notting". PAINT use to have this, there was lengthy debate and it was removed. Paul would be better at summarizing why, but the gist is that if a function has been lost then the correct thing to say (from an evolutionary standpoint) is that it does not have that function any longer.

In short the blocking vs. notting gets into questions of the evolutionary model behind the curator's process.

**Comment by selewis** [comment]
P.S. why not just annotate to the more general term at the top (closer to root) and then add the more specific classes at the appropriate branch points?
